File: app/src/main/java/com/gushenge/testdemo/TaobaoNews/NewsAdapter.java
package com.gushenge.testdemo.TaobaoNews;

import android.content.Context;
import android.graphics.Color;
import android.util.TypedValue;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import java.util.Random;

import com.gushenge.testdemo.R;

public class NewsAdapter extends BaseAdapter implements AutoScrollView.AutoScroll {
    private Random random = new Random();
    private LayoutInflater mLayoutInflater;
    @Override
    public int getCount() {
        return 10;
    }

    @Override
    public Object getItem(int position) {
        return null;
    }

    @Override
    public long getItemId(int position) {
        return 0;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        ViewHolder viewHolder;
        if (convertView == null) {
            if (mLayoutInflater == null) {
                mLayoutInflater = LayoutInflater.from(parent.getContext());
            }
            convertView = mLayoutInflater.inflate(R.layout.news_item, parent,false);
            viewHolder = new ViewHolder();
            viewHolder.tvTitle = (TextView) convertView.findViewById(R.id.tv_title);
            convertView.setTag(viewHolder);
        }else {
            viewHolder = (ViewHolder) convertView.getTag();
        }
        convertView.setBackgroundColor(Color.argb(100, random.nextInt(255), random.nextInt(255), random.nextInt(255)));
        viewHolder.tvTitle.setText("第"+position + "个item");
        return convertView;
    }

    /*传入item高度*/
    @Override
    public int getListItemHeight(Context context) {
        return (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 50, context.getResources().getDisplayMetrics());
    }
    /*传入展示数量*/
    @Override
    public int getImmovableCount() {
        return 2;
    }

    class ViewHolder{
        public TextView tvTitle;
    }
}

File: app/src/main/kotlin/com/gushenge/testdemo/TaobaoNews/NewsAdapter.kt
package com.gushenge.testdemo.TaobaoNews

import android.content.Context
import android.graphics.Color
import android.util.TypedValue
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView

import java.util.Random

import com.gushenge.testdemo.R
 class NewsAdapter(val context:Context) : BaseAdapter(), AutoScrollView.AutoScroll {
     override val immovableCount: Int
         get() = 2
     private val random = Random()
    private var mLayoutInflater: LayoutInflater? = null
    override fun getCount(): Int {
        return 10
    }

    override fun getItem(position: Int): Any? {
        return null
    }

    override fun getItemId(position: Int): Long {
        return 0
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val view:View
        val viewHolder: ViewHolder
        if (convertView == null) {
            view = LayoutInflater.from(context).inflate(R.layout.news_item, parent, false)
            viewHolder = ViewHolder()
            viewHolder.tvTitle = view.findViewById<TextView>(R.id.tv_title)
            view.tag = viewHolder
        } else {
            view = convertView
            viewHolder = convertView.tag as ViewHolder
        }
        viewHolder.tvTitle!!.text = "第" + position + "个item"
        return view
    }

    /*传入item高度*/
    override fun getListItemHeight(context: Context): Int {
        return TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 50f, context.resources.displayMetrics).toInt()
    }


    internal inner class ViewHolder {
        var tvTitle: TextView? = null
    }
}
